messaging/src/main/java/org/springframework/security/messaging/web/csrf/XorCsrfTokenUtils.java

```diff
@@ -56,13 +56,16 @@ static String getTokenValue(String actualToken, String token) {
 		System.arraycopy(actualBytes, randomBytesSize, xoredCsrf, 0, tokenSize);
 
 		byte[] csrfBytes = xorCsrf(randomBytes, xoredCsrf);
-		return Utf8.decode(csrfBytes);
+		return (csrfBytes != null) ? Utf8.decode(csrfBytes) : null;
 	}
 
-	private static byte[] xorCsrf(byte[] randomBytes, byte[] csrfBytes) {
+	static byte[] xorCsrf(byte[] randomBytes, byte[] csrfBytes) {
+		if (csrfBytes.length < randomBytes.length) {
+			return null;
+		}
 		int len = Math.min(randomBytes.length, csrfBytes.length);
```

Comment:
Hi,
Isn't the value of `randomBytes.length` will always be smaller or equals to the value of `csrfBytes.length`?
If so you can remove the usage of the min fucntion and just call the value

Reply:
Thanks for your comments on this PR @roysav13! I will be addressing this comment in a separate fix. See gh-15184.